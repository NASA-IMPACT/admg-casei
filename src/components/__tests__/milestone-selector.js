import React from "react"
import renderer from "react-test-renderer"
import MilestoneSelector from "../timeline/milestone-selector"

const testEvents = [
  {
    campaign: "c0dc0d63-e32a-4fdc-b1ac-4f0e8a5c7754",
    end: "2012-09-20",
    collectionPeriods: [
      { id: "8864fcb5-444c-4fd7-826b-0240bcf124cd" },
      { id: "de7a9541-a67d-4ba3-ae77-5a8856bb549e" },
      { id: "545d68d7-9787-4435-956d-9bd663bb2224" },
    ],
    id: "f2c42b8a-2c06-407c-9345-aa9824ac65f5",
    longname: "Western US Fall 2012",
    region: [],
    shortname: "AirMOSS_dep_2012a",
    start: "2012-09-18",
  },
  {
    campaign: "c0dc0d63-e32a-4fdc-b1ac-4f0e8a5c7754",
    end: "2012-10-20",
    collectionPeriods: [
      { id: "1e2f5b99-78f6-4b40-b634-67957fa26311" },
      { id: "6ca9f494-7948-4b42-bf5c-956b0d7cf0b9" },
    ],
    id: "9f98f6f9-ea1b-4ef7-a891-8fb175bf1df5",
    longname: "Canada, Oregon Fall 2012",
    region: [],
    shortname: "AirMOSS_dep_2012b",
    start: "2012-10-02",
  },
]
const testId = "test id"
const testAction = () => {}

describe("Timeline", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <MilestoneSelector
          events={testEvents}
          timelineAction={testAction}
          activeMilestone={testId}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
