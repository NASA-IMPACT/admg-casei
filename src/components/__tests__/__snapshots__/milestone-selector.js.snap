// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline renders correctly 1`] = `
.c0 {
  display: grid;
  grid-template-columns: 30px 120px;
  grid-template-areas: "icon title" "icon subTitle" "icon flights";
}

.c1 {
  margin: 0;
  font-size: small;
  text-align: start;
}

.c2 {
  margin: 0;
  font-size: x-small;
  text-align: start;
}

<section
  data-cy="milestone-timeline"
  style={
    Object {
      "borderTop": "1px solid gray",
      "whiteSpace": "nowrap",
    }
  }
>
  <ol
    style={
      Object {
        "display": "flex",
        "height": "200px",
        "margin": "0",
        "maxWidth": "100%",
        "overflowX": "scroll",
        "padding": "8rem 0 2rem 0",
      }
    }
  >
    <li
      style={
        Object {
          "background": "#fff",
          "display": "flex",
          "flex": "1 0 25%",
          "height": "2px",
          "listStyleType": "none",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "background": "hsla(0,0%,100%,0.9)",
            "bottom": "0rem",
            "height": "2rem",
            "position": "absolute",
            "width": "1px",
          }
        }
      />
      <button
        data-cy="milestone-timeline-card"
        onClick={[Function]}
        style={
          Object {
            "background": "hsla(0,0%,100%,0.9)",
            "border": "none",
            "bottom": "2rem",
            "color": "black",
            "opacity": 0.7,
            "padding": ".5rem",
            "position": "absolute",
            "width": "12rem",
          }
        }
      >
        <div
          className="c0"
          data-cy="milestone-timeline-card-content"
        >
          <div
            style={
              Object {
                "gridArea": "icon",
              }
            }
          >
            <svg
              height="30"
              viewBox="0 0 145 140"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.378 70.315l1.767 1.767-4.879 4.877 3.301.001v2.5H0v-7.567h2.5v3.298l4.878-4.876zM61.917 3.45L76.01 17.543 58.835 34.718l-6.163-6.162-2.308 2.308 8.799 8.798c3.203-.678 5.87-.124 7.605 1.613l.885.885L55.79 54.022l1.107 1.108a3.986 3.986 0 014.593.74 3.982 3.982 0 010 5.62 3.957 3.957 0 01-2.81 1.163 3.962 3.962 0 01-2.81-1.163 3.985 3.985 0 01-.74-4.593l-1.107-1.107-11.866 11.865-.882-.885c-1.775-1.775-2.26-4.498-1.613-7.606l-8.801-8.801-2.308 2.308 6.165 6.164L17.542 76.01 3.45 61.917l17.175-17.175 6.161 6.162 2.308-2.308-7.279-7.278L26.35 26.35l14.965-4.535 7.28 7.28 2.308-2.308-6.163-6.162L61.917 3.45zM20.624 48.278l-13.64 13.64 10.558 10.557 13.64-13.64-4.397-4.396L15.037 66.19l-1.767-1.767L25.019 52.67l-4.394-4.393zm49.61 14.392v2.5l-3.298-.001 4.876 4.876-1.768 1.767-4.878-4.877v3.3h-2.5V62.67h7.568zm-8.572-20.78c-.978 0-2.078.185-3.26.555-3.223 1.01-6.666 3.235-9.693 6.265-5.33 5.332-7.97 11.875-6.392 15.25L63.95 42.327c-.645-.292-1.415-.437-2.288-.437zM58.68 57.208c-.377 0-.755.145-1.042.43a1.48 1.48 0 000 2.084c.577.578 1.51.578 2.084 0a1.476 1.476 0 000-2.084 1.479 1.479 0 00-1.042-.43zM40.608 24.642l-12.253 3.713L24.64 40.61l15.84 15.838c.278-.719.61-1.448.991-2.183l-8.753-8.755 1.767-1.77 8.292 8.296a31.898 31.898 0 014.166-5.093c2.94-2.94 6.273-5.202 9.506-6.46l-15.841-15.84zM30.684 39.94l2.31 2.313-1.767 1.767-2.31-2.31 1.767-1.77zM61.917 6.985l-13.642 13.64 4.395 4.394L64.42 13.27l1.77 1.768-11.75 11.75 4.395 4.395 13.64-13.64L61.918 6.985zM79.46 0v7.567h-2.5V4.265l-4.877 4.88-1.768-1.767L75.19 2.5h-3.297V0h7.567z"
                fill="#0c1520"
                fillRule="evenodd"
              />
            </svg>
          </div>
          <p
            className="c1"
            style={
              Object {
                "gridArea": "title",
              }
            }
          >
            Western US Fall 2012
          </p>
          <p
            className="c2"
            style={
              Object {
                "gridArea": "subTitle",
              }
            }
          >
            missing details
          </p>
          <p
            className="c2"
            style={
              Object {
                "gridArea": "flights",
              }
            }
          >
            3 Flights
          </p>
        </div>
      </button>
      <p
        style={
          Object {
            "color": "hsla(0,0%,100%,0.9)",
            "height": "40px",
            "position": "absolute",
            "top": "1rem",
            "width": "100%",
          }
        }
      >
        2012-09-20
      </p>
    </li>
    <li
      style={
        Object {
          "background": "#fff",
          "display": "flex",
          "flex": "1 0 25%",
          "height": "2px",
          "listStyleType": "none",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "background": "hsla(0,0%,100%,0.9)",
            "bottom": "0rem",
            "height": "2rem",
            "position": "absolute",
            "width": "1px",
          }
        }
      />
      <button
        data-cy="milestone-timeline-card"
        onClick={[Function]}
        style={
          Object {
            "background": "hsla(0,0%,100%,0.9)",
            "border": "none",
            "bottom": "2rem",
            "color": "black",
            "opacity": 0.7,
            "padding": ".5rem",
            "position": "absolute",
            "width": "12rem",
          }
        }
      >
        <div
          className="c0"
          data-cy="milestone-timeline-card-content"
        >
          <div
            style={
              Object {
                "gridArea": "icon",
              }
            }
          >
            <svg
              height="30"
              viewBox="0 0 145 140"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.378 70.315l1.767 1.767-4.879 4.877 3.301.001v2.5H0v-7.567h2.5v3.298l4.878-4.876zM61.917 3.45L76.01 17.543 58.835 34.718l-6.163-6.162-2.308 2.308 8.799 8.798c3.203-.678 5.87-.124 7.605 1.613l.885.885L55.79 54.022l1.107 1.108a3.986 3.986 0 014.593.74 3.982 3.982 0 010 5.62 3.957 3.957 0 01-2.81 1.163 3.962 3.962 0 01-2.81-1.163 3.985 3.985 0 01-.74-4.593l-1.107-1.107-11.866 11.865-.882-.885c-1.775-1.775-2.26-4.498-1.613-7.606l-8.801-8.801-2.308 2.308 6.165 6.164L17.542 76.01 3.45 61.917l17.175-17.175 6.161 6.162 2.308-2.308-7.279-7.278L26.35 26.35l14.965-4.535 7.28 7.28 2.308-2.308-6.163-6.162L61.917 3.45zM20.624 48.278l-13.64 13.64 10.558 10.557 13.64-13.64-4.397-4.396L15.037 66.19l-1.767-1.767L25.019 52.67l-4.394-4.393zm49.61 14.392v2.5l-3.298-.001 4.876 4.876-1.768 1.767-4.878-4.877v3.3h-2.5V62.67h7.568zm-8.572-20.78c-.978 0-2.078.185-3.26.555-3.223 1.01-6.666 3.235-9.693 6.265-5.33 5.332-7.97 11.875-6.392 15.25L63.95 42.327c-.645-.292-1.415-.437-2.288-.437zM58.68 57.208c-.377 0-.755.145-1.042.43a1.48 1.48 0 000 2.084c.577.578 1.51.578 2.084 0a1.476 1.476 0 000-2.084 1.479 1.479 0 00-1.042-.43zM40.608 24.642l-12.253 3.713L24.64 40.61l15.84 15.838c.278-.719.61-1.448.991-2.183l-8.753-8.755 1.767-1.77 8.292 8.296a31.898 31.898 0 014.166-5.093c2.94-2.94 6.273-5.202 9.506-6.46l-15.841-15.84zM30.684 39.94l2.31 2.313-1.767 1.767-2.31-2.31 1.767-1.77zM61.917 6.985l-13.642 13.64 4.395 4.394L64.42 13.27l1.77 1.768-11.75 11.75 4.395 4.395 13.64-13.64L61.918 6.985zM79.46 0v7.567h-2.5V4.265l-4.877 4.88-1.768-1.767L75.19 2.5h-3.297V0h7.567z"
                fill="#0c1520"
                fillRule="evenodd"
              />
            </svg>
          </div>
          <p
            className="c1"
            style={
              Object {
                "gridArea": "title",
              }
            }
          >
            Canada, Oregon Fall 2012
          </p>
          <p
            className="c2"
            style={
              Object {
                "gridArea": "subTitle",
              }
            }
          >
            missing details
          </p>
          <p
            className="c2"
            style={
              Object {
                "gridArea": "flights",
              }
            }
          >
            2 Flights
          </p>
        </div>
      </button>
      <p
        style={
          Object {
            "color": "hsla(0,0%,100%,0.9)",
            "height": "40px",
            "position": "absolute",
            "top": "1rem",
            "width": "100%",
          }
        }
      >
        2012-10-20
      </p>
    </li>
  </ol>
</section>
`;
